pkgs/applications/misc/megasync/default.nix

```diff
@@ -0,0 +1,143 @@
+{ lib
+, stdenv
+, autoconf
+, automake
+, bash
+, bashInteractive
+, c-ares
+, cryptopp
+, curl
+, doxygen
+, fetchFromGitHub
+, hicolor-icon-theme
+, libmediainfo
+, libraw
+, libsodium
+, libtool
+, libuv
+, libzen
+, lsb-release
+, makeDesktopItem
+, pkgconfig
+, qt5
+, sqlite
+, swig
+, unzip
+, wget
+, enableFFmpeg   ? true, ffmpeg  ? ffmpeg
+}:
+
+assert enableFFmpeg   -> ffmpeg != null;
+
+stdenv.mkDerivation rec {
+  name = "megasync-${version}";
+  version = "4.1.1.0";
+
+  src = fetchFromGitHub {
+    owner = "meganz";
+    repo = "MEGAsync";
+    rev = "v${version}_Linux";
+    sha256 = "0lc228q3s9xp78dxjn22g6anqlsy1hi7a6yfs4q3l6gyfc3qcxl2";
+    fetchSubmodules = true;
+  };
+
+  desktopItem = makeDesktopItem {
```

Comment:
The `desktopItem` should be outside the derivation in a let block in a pattern like this:
```
let 
  desktopItem = ... {
  };
in stdenv.mkDerivation { ...

Reply:
Using a provided one now.